aten/src/ATen/core/TensorImpl.h

```diff
@@ -229,6 +229,28 @@ struct CAFFE2_API TensorImpl : public c10::intrusive_ptr_target {
     return is_contiguous_;
   }
 
+  bool is_sparse() const {
+    // NB: This method is not virtual and avoid dispatches for performance reasons.
+    const auto& tid = type_id();
```

Comment:
This is harmless, in the sense that the temporary will get its scope extended for the reference, but it's totally unnecessary, because TensorTypeId is a value object and should be treated as such.